### PR TITLE
deployment: docker image and build commands

### PR DESCRIPTION
**Pull request by travisgroth**
## Summary

- Add release Docker image
- Add build steps to `Makefile`
- Add main branch Docker image 

## Related issues

n/a

## Checklist

- [ ] reference any related issues
- [ ] updated docs
- [ ] updated unit tests
- [ ] updated UPGRADING.md
- [x] add appropriate tag (`improvement` / `bug` / etc)
- [x] ready for review

### CODE REVIEW

File: goreleaser.yaml [hunk 1]
```diff
@@ -100,7 +100,7 @@ nfpms:
     package_name: pomerium-cli
     vendor: Pomerium, Inc.
     homepage: https://www.pomerium.com
-    description: Identity Aware Proxy
+    description: Local Client for Identity Aware Proxy
```
Comment (desimone) [inline goreleaser.yaml:103]:
```suggestion
    description: Local client for Pomerium, an identity aware proxy.
```

File: Dockerfile.release [hunk 2]
```diff
@@ -0,0 +1,6 @@
+ARG ARCH
+
+FROM gcr.io/distroless/base-debian10:latest-${ARCH}
```
Comment (desimone) [inline Dockerfile.release:3]: Should we use `gcr.io/distroless/static` here?  I guess my broader question is, do we need cgo here?

**Review by desimone — APPROVED** [review]

File: Dockerfile.release (hunk 2 above)
Comment (travisgroth) [reply Dockerfile.release:3]: 🤔  we can probably do static since we aren't bringing envoy with us here.

File: Makefile [hunk 3]
```diff
@@ -17,3 +20,16 @@ tidy:
...
+	mkdir -p ./bin
+	go build -o bin/pomerium-cli cmd/*.go
+
+.PHONY: clean
+clean:
+	rm -f bin/*
+
+.PHONY: snapshot
+snapshot:
+	APPARITOR_GITHUB_TOKEN=foo goreleaser release --snapshot --rm-dist
```
Comment (desimone) [inline Makefile:35]:
```suggestion
	APPARITOR_GITHUB_TOKEN=foo goreleaser release --snapshot --rm-dist
```

foo? :)

File: goreleaser.yaml [hunk 4]
```diff
@@ -17,7 +17,7 @@ before:
 
 builds:
   - id: pomerium-cli
-    main: ./cmd/main.go
+    main: ./cmd
```
Comment (desimone) [inline goreleaser.yaml:20]: I hit an issue with a root command actually working. See #11 ; maybe you found a way around it?

File: .github/workflows/release.yaml [hunk 5]
```diff
@@ -67,3 +67,15 @@ jobs:
...
+        id: latestTag
+        run: |
+          LATEST_TAG=$(git tag | grep -vi 'rc' | sort --version-sort | tail -1)
+          echo "::set-output name=tag::${LATEST_TAG}"
+
+      - name: Publish latest tag
+        if: "steps.latestTag.outputs.tag == steps.tagName.outputs.tag"
+        run: |
+          docker manifest create -a pomerium/cli:latest pomerium/cli:amd64-${{ steps.tagName.outputs.tag }} pomerium/cli:arm64v8-${{ steps.tagName.outputs.tag }}
+          docker manifest push pomerium/pomerium:latest
```
Comment (desimone) [inline .github/workflows/release.yaml:81]:
```suggestion
          docker manifest push pomerium/cli:latest
```
😌

File: .github/workflows/release.yaml [hunk 6]
```diff
@@ -67,3 +67,15 @@ jobs:
           for pkg in $(echo $DEBS); do
             cloudsmith push deb pomerium/pomerium/debian/any-version $pkg
```
Comment (desimone) [inline .github/workflows/release.yaml:68]:
```suggestion
            cloudsmith push deb pomerium/cli/debian/any-version $pkg
```

?

**Review by desimone — CHANGES_REQUESTED** [review]

File: Makefile (hunk 3 above)
Comment (travisgroth) [reply Makefile:35]: So it doesn't yell at you for not having a token set.

File: goreleaser.yaml (hunk 4 above)
Comment (travisgroth) [reply goreleaser.yaml:20]: It had to be built with `cmd/*.go`, which is what the syntax here does implicitly.

File: Dockerfile.release (hunk 2 above)
Comment (travisgroth) [reply Dockerfile.release]: 👍

File: .github/workflows/release.yaml (hunk 6 above)
Comment (travisgroth) [reply .github/workflows/release.yaml:68]: That's the repo path, not package name.

**Review by calebdoxsey — APPROVED** [review]

**Comment by travisgroth** [comment]
Rebased onto #11

**Review by desimone — APPROVED** [review]

**Pull request merged**